src/edu/ccsu/cs505/hw2/BankAccount.java

```diff
@@ -0,0 +1,84 @@
+package edu.ccsu.cs505.hw2;
+
+/**
+ * @author Julie Shevchenko
+ * BankAccount class, which extends Account.
+ * Overrides setNextChain and deduct methods 
+ **/
+public class BankAccount extends Account{
+    private String accountName;
+    private double accountBalance;
+    private Account chainAccount;
+
+    /** 
+     * constructor for BankAccount
+     * takes name, balance, and chain and sets accountName, accountBalance, and chainAccount variables to those values
+     * @param name accountName
+     * @param balance accountBalance
+     * @param chain chainAccount
+     **/
+    public BankAccount(String name, double balance, Account chain) {
+        accountName = name;
+        accountBalance = balance;
+        chainAccount = chain;
+    }
+
+    /**
+     * @see setNextChain(Account nextChain) Javadoc in Account class
+     **/
+     @Override
+    public void setNextChain(Account nextChain) {
+        this.chainAccount = nextChain;
+    }
+
+    /**
+     * @see deduct(double amount) throws InsufficientFundsException Javadoc in Account class
+     **/
+    @Override
+    public void deduct(double amount) throws InsufficientFundsException {
+        if (amount <= accountBalance) {
+            double remaining = accountBalance - amount;
+            System.out.println(accountName + " paid $" + amount + ". $" + remaining + " remaining in account.");
+        } else {
+            double remaining = amount - accountBalance;
+            System.out.println(accountName + " paid $" + accountBalance + ". $" + remaining + " still needs to be paid.");
+            this.chainAccount.deduct(remaining);
```

Comment:
Logic error where if exception is thrown the balance is left in an inconsistent state - zero even though request could not be fulfilled.